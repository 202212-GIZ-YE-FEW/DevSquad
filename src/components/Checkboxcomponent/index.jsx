import { useState } from "react";
const Inputcomponent = ({
    textColor = "text-white",
    textColorChecked = "text-primary-orange",
    borderColor = "border-primary-orange",
    borderRightAndDown = " ",
    font = "font-Rubik",
    height = "h-32",
    title,
    checked,
    ...props
}) => {
    const defaultChecked = checked ? checked : false;
    const [isChecked, setIsChecked] = useState(defaultChecked);
    return (
        <div>
            <label>
                <input
                    type='checkbox'
                    checked={isChecked}
                    onChange={() => setIsChecked((prev) => !prev)}
                    className={isChecked ? "checked" : ""}
                    {...props}
                    hidden
                />
                <span
                    className={
                        isChecked
                            ? `checked flex items-center justify-center text-center ${textColorChecked} border ${borderColor} ${borderRightAndDown} p-3 rounded ${height}  ${font} font-medium sm:text-base text-sm`
                            : `flex items-center justify-center ${textColor} text-center border ${borderColor} ${borderRightAndDown} bg-primary-orange p-3 rounded ${height} ${font} font-medium sm:text-base text-sm`
                    }
                >
                    {title}
                </span>
            </label>
        </div>
    );
};

export default Inputcomponent;
