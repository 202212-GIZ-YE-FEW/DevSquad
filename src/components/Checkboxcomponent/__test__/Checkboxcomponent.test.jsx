import renderer from "react-test-renderer";

import Checkboxcomponent from "../index";

it("renders correctly", () => {
    const tree = renderer.create(<Checkboxcomponent />).toJSON();
    expect(tree).toMatchSnapshot();
});
