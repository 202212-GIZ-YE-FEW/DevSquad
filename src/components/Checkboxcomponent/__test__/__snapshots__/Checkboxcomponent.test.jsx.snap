// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <label
    className="flex md:block"
  >
    <input
      className="undefined"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="undefined"
    />
  </label>
</div>
`;
